python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -567,7 +569,25 @@ def my_function():
     def inner(fn: Callable[..., Any]) -> AssetsDefinition:
         op_name = name or fn.__name__
         asset_ins = build_asset_ins(fn, ins or {}, deps=_make_asset_keys(upstream_asset_deps))
-        asset_outs = build_asset_outs(outs)
+
+        if outs and assets:
+            raise DagsterInvalidDefinitionError("Must specify only outs or assets but not both.")
+        elif outs:
+            asset_outs = build_asset_outs(outs)
+        elif assets:
+            asset_outs = {}
+            for asset_node in assets:
+                output_name = asset_node.asset_key.path[-1]
+                # output naming strategy? using last path entry
```

Comment:
I think it is important to build some output name that guarantees uniqueness. If we don't want super long names in the case of a long asset key, we could do a short hash or just append a number to it or something.